src/App.jsx

```diff
@@ -14,33 +15,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const GoodList = ({ goodsArray }) => {
+  return (
+    <ul>
+      {goodsArray.map(item => (
+        <li key={item} data-cy="Good">
+          {item}
+        </li>
+      ))}
+    </ul>
+  );
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const sorting = (array, isReversed, sortField) => {
+  let sortedArray = [...array];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField === 'alphabetically') {
+    sortedArray.sort((good1, good2) => good1.localeCompare(good2));
+  } else if (sortField === 'length') {
+    sortedArray.sort((good1, good2) => good1.length - good2.length);
+  } else if (sortField === 'reset') {
+    sortedArray = array;
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    sortedArray.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return sortedArray;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isVisibleReset, setIsVisibleReset] = useState(false);
+  const sortByAlphabetically = () => {
+    setSortField('alphabetically');
+    setIsVisibleReset(true);
+  };
+
+  const sortByLength = () => {
+    setSortField('length');
+    setIsVisibleReset(true);
+  };
+
+  const reverseButton = () => {
+    setIsReversed(prev => {
+      const nextReversedState = !prev;
+
+      if (!nextReversedState) {
+        setIsVisibleReset(false);
+      } else {
+        setIsVisibleReset(true);
+      }
```

Comment:
The logic for setting `isVisibleReset` in the `reverseButton` function could be simplified. Since `isVisibleReset` is set to true whenever the reverse state changes, you might not need to toggle it based on the `nextReversedState`.